engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobTimeoutTrigger.java

```diff
@@ -57,9 +57,8 @@ public void onPaused() {
 
   @Override
   public void onResumed() {
-    if (shouldReschedule) {
-      scheduleDeactivateTimedOutJobsTask();
-    }
+    shouldReschedule = true;
+    scheduleDeactivateTimedOutJobsTask();
```

Comment:
🔧 @korthout I think you mentioned directly calling `deactivateTimedOutJobs`. Not sure whether this is easily possible, since we need the taskResultBuilder. We could schedule it for Zero delay if you want.

https://github.com/camunda/zeebe/issues/13190#issuecomment-1631880802

This case is btw harder to test, that it is immediately executed, instead of after the delay, but might be enough our current next test. Let me know what you think.

Reply:
I was not specific enough. I did not specifically mean directly calling it or scheduling it again. Simply that we should make a starting pass again. So, I'm happy with the solution you proposed.